src/vmsa/mod.rs

```diff
@@ -305,33 +306,19 @@ pub struct Vmsa {
     x87_state_gpa: u64,
 }
 
-/// An enum that encompasses all valid return types of VMSA reading/writing and
-/// serialization/deserialization.
-pub enum VmsaRWResult {
-    /// An error occurred in reading/writing a VMSA struct.
-    IoErr(io::Error),
+impl codicon::Decoder<()> for Vmsa {
+    type Error = std::io::Error;
```

Comment:
I see, we're able to remove the use of `bincode::Error` by instead having the serialization/deserialization return an `io::Error` instead. This is much cleaner than the previous approach.